src/App.tsx

```diff
@@ -1,24 +1,186 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { FilteredBy, filteredTodoList } from './helpers';
+import { Errors } from './components/Errors/Errors';
+import { ErrorType } from './types/ErorTypes';
 
-const USER_ID = 0;
+const USER_ID = 12036;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilteredBy>(FilteredBy.DefaultType);
+  const [errorType, setErorType] = useState<ErrorType | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoadingTodo, setIsLoadingTodo] = useState<Todo | null>(null);
+
+  const todosToView = useMemo(
+    () => filteredTodoList(todos, filterBy),
+    [todos, filterBy],
+  );
+
+  const deleteTodo = useCallback(
+    async (todoId: number) => {
+      try {
+        const todoToRemove = todos.find((todo) => todo.id === todoId) || null;
+
+        setIsLoadingTodo(todoToRemove);
+
+        await todoService.removeTodo(todoId);
+
+        setTodos((currentTodos) => currentTodos.filter(
+          (todo) => todo.id !== todoId,
+        ));
+      } catch (error) {
+        setErorType(ErrorType.DeleteError);
+      }
+    },
+    [todos, setIsLoadingTodo, setTodos, setErorType],
+  );
+
+  const addTodo = useCallback(
+    async (todoTitle: string) => {
+      try {
+        setIsLoading(true);
+        const newTempTodo: Todo = {
+          id: 0,
+          userId: USER_ID,
+          title: todoTitle,
+          completed: false,
+        };
+
+        const addedTodo = await todoService.addTodo({
+          title: todoTitle,
+          completed: false,
+          userId: USER_ID,
+        });
+
+        setTempTodo(newTempTodo);
```

Comment:
```suggestion
        setTempTodo(newTempTodo);
        
        const addedTodo = await todoService.addTodo({
          title: todoTitle,
          completed: false,
          userId: USER_ID,
        });
```
You should show temp todo before the request succeeded, as now you show it only after you have a new todo and then delete it